src/Microsoft.Identity.Web/Resource/TokenAuthorizationRules.cs

```diff
@@ -0,0 +1,150 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Claims;
+
+namespace Microsoft.Identity.Web.Resource
+{
+    /// <summary>
+    /// Rules to authorize an access token in a web API, depending on the
+    /// values of the <see cref="ITokenAuthorizationOptions"/>.
+    /// </summary>
+    public class TokenAuthorizationRules
+    {
+        /// <summary>
+        /// Validates the authorization of the token.
+        /// </summary>
+        /// <param name="tokenAuthorizationOptions">Merged options to direct the authorization.</param>
+        /// <param name="claimsPrincipal">Claims in the token.</param>
+        public static void ValidateApiAuthorization(
+            ITokenAuthorizationOptions tokenAuthorizationOptions,
+            ClaimsPrincipal claimsPrincipal)
+        {
+            if (tokenAuthorizationOptions is null)
+            {
+                throw new ArgumentNullException(nameof(tokenAuthorizationOptions));
+            }
+
+            if (claimsPrincipal is null)
+            {
+                throw new ArgumentNullException(nameof(claimsPrincipal));
+            }
+
+            // Does the web API disallow app only tokens?
+            if (!tokenAuthorizationOptions.ApiAllowsAppOnlyTokens)
+            {
+                VerifyClaimDoesNotExistOrDoesNotEquals(claimsPrincipal, ClaimConstants.Idtyp, "app", IDWebErrorMessage.ApiAllowsAppOnlyTokensCantBeComputed, IDWebErrorMessage.ApiDoesNotAllowsAppOnlyTokens);
+            }
+
+            // Does the web API disallow guest accounts?
+            if (!tokenAuthorizationOptions.ApiAllowsGuestAccounts)
+            {
+                VerifyClaimEquals(claimsPrincipal, ClaimConstants.Acct, "0", IDWebErrorMessage.ApiAllowsGuestAccountsCantBeComputed, IDWebErrorMessage.ApiDoesNotAllowsGuestAccounts);
```

Comment:
Validated, correct value is indeed 0.

Reply:
Should this value (and others, such as "app") maybe be defined in a constant?